pkg/util/slice/slice.go

```diff
@@ -17,55 +17,14 @@ limitations under the License.
 // Package slice provides utility methods for common operations on slices.
 package slice
 
-import (
-	"sort"
-)
+import "slices"
 
-// CopyStrings copies the contents of the specified string slice
-// into a new slice.
-func CopyStrings(s []string) []string {
-	if s == nil {
-		return nil
-	}
-	c := make([]string, len(s))
-	copy(c, s)
-	return c
-}
-
-// SortStrings sorts the specified string slice in place. It returns the same
-// slice that was provided in order to facilitate method chaining.
-func SortStrings(s []string) []string {
-	sort.Strings(s)
-	return s
-}
-
-// ContainsString checks if a given slice of strings contains the provided string.
-// If a modifier func is provided, it is called with the slice item before the comparation.
-func ContainsString(slice []string, s string, modifier func(s string) string) bool {
-	for _, item := range slice {
-		if item == s {
-			return true
-		}
-		if modifier != nil && modifier(item) == s {
-			return true
-		}
-	}
-	return false
-}
-
-// RemoveString returns a newly created []string that contains all items from slice that
+// Remove returns a newly created []string that contains all items from slice that
 // are not equal to s and modifier(s) in case modifier func is provided.
-func RemoveString(slice []string, s string, modifier func(s string) string) []string {
-	newSlice := make([]string, 0)
-	for _, item := range slice {
-		if item == s {
-			continue
-		}
-		if modifier != nil && modifier(item) == s {
-			continue
-		}
-		newSlice = append(newSlice, item)
-	}
+func Remove(finalizers []string, toRemove string) []string {
```

Comment:
Would it make sense to make this function generic?

Reply:
It would, originally didn't do it since it was designed for finalizers, but now it can be 😄